### PR TITLE
[JBPM-9184] Subprocess excepcion handling tx rollback causes main process to fail

### PR DESCRIPTION
**Pull request by elguardian**
test case provide by Gonzalo

### CODE REVIEW

**Comment by elguardian** [comment]
depends on https://github.com/kiegroup/drools/pull/2945

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        KieServerDeployer.buildAndDeployCommonMavenParent();
+        KieServerDeployer.buildAndDeployMavenProjectFromResource("/kjars-sources/exception-handling");
+
+        kieContainer = KieServices.Factory.get().newKieContainer(releaseId);
+
+        createContainer(CONTAINER_ID, releaseId, PPI_RUNTIME_STRATEGY);
+    }
+
+    @Test
+    public void test1() throws Exception {
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java:51]: Please use more descriptive method name.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        try {
+            processInstanceId = processClient.startProcess(CONTAINER_ID, "ExceptionHandling.ScriptMainProcess");
+
+            assertNotNull(processInstanceId);
+            assertTrue(processInstanceId.longValue() > 0);
+
+            ProcessInstance a = processClient.getProcessInstance(CONTAINER_ID, processInstanceId);
+            assertNotNull(a);
+            processInstanceId = null;
+        } catch (Exception e) {
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java:62]: Please remove the catch

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java:2]: 2020

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java (hunk 2 above)
Comment (afalhambra) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java:62]: +1

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java (hunk 1 above)
Comment (afalhambra) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java:51]: +1

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java (hunk 3 above)
Comment (afalhambra) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java:2]: +1

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java (hunk 2 above)
Comment (elguardian) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java]: done

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java (hunk 3 above)
Comment (elguardian) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java]: done

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java (hunk 1 above)
Comment (elguardian) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java]: done

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java (hunk 3 above)
Comment (elguardian) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java]: done

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java (hunk 2 above)
Comment (elguardian) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ExceptionHandlingIntegrationTest.java]: done

**Comment by mareknovotny** [comment]
Jenkins retest this please

**Comment by elguardian** [comment]
Jenkins retest this please.

**Pull request merged**